web/src/app/UserTasks.tsx

```diff
@@ -0,0 +1,29 @@
+"use client";
+
+import { useAuth } from "./hooks/useAuth";
+
+export default function UserTasks() {
+  const { loading, isAuthed, userData } = useAuth();
+
+  if (loading) return <p>Loading…</p>;
+
+  if (!isAuthed) {
+    return (
+      <a href={`/oauth2/start?rd=${encodeURIComponent(window.location.pathname)}`} className="btn--login" id="login">
+        Login with UCL SSO
+      </a>
+    );
+  }
```

Comment:
This can be extracted into a custom component in a follow-up PR. I'm also not totally sure if the custom redirect is working yet.

Reply:
It is defined elsewhere so we don’t strictly need to define it again. The idea here would be that the redirect could be dynamic based on which page the user was asked to login from. So a slightly nicer UX experience.